src/server/game/AI/ScriptedAI/ScriptedCreature.cpp

```diff
@@ -123,6 +123,17 @@ void SummonList::DoActionImpl(int32 action, StorageType const& summons)
     }
 }
 
+std::list<ObjectGuid> SummonList::GetGUIDs() const
```

Comment:
Is there any way making this return some sort of generic container like IEnumerable in C#?

Reply:
What for?

Also bear in mind what the impl of this method is actually doing: copying the values and referencing em as copy.
Also also, better use vectors, GuidVector is the typename IIRC

Reply:
I am not sure how you are guys doing, isn't methods supposed to be at most generic as possible? 
Yes, I will change it to vector.

Reply:
I'm afraid the most generic solution you can currently achieve in C++ is not returning anything from the method, but instead make it a template accepting a parameter of arbitrary iterator type, and filling it like `itr++ = guid`. That way users of the method can call it like this:
```cpp
std::vector<ObjectGuid> v;
GetGUIDs(std::back_inserter(v));
std::list<ObjectGuid> l;
GetGUIDs(std::back_inserter(l));
std::set<ObjectGuid> s;
GetGUIDs(std::inserter(s));
```
But I'm not saying you should do it, though. 😄

C#'s IEnumerable, besides being an actual interface (which don't really exist in C++), has an additional benefit of allowing lazy coroutine-like execution: if you write a function that returns IEnumerable and you return data from it via `yield return`, the compiler breaks your function into multiple pieces under the hood (e.g. initialization, loop, conclusion) and only executes each loop iteration when the next element of IEnumerable is requested. That way if you break your loop after the first iteration - only the code until the first `yield return` will end up being executed and nothing else. C++ doesn't have anything like that currently, and while you could certainly code it yourself (by making a custom iterator that calls the appropriate method in `operator++`), you'd have to deal with breaking the function into pieces yourself, which is inconvenient at best (and completely unusable at worst). Coroutine TS [could make it much more pleasant](https://kirit.com/How%20C%2B%2B%20coroutines%20work/Yielding%20Generators) though, when it finally gets included in C++ standard.